File: android/app/src/main/java/com/texasgamer/zephyr/util/StringUtils.java
package com.texasgamer.zephyr.util;

import androidx.annotation.Nullable;

/**
 * String utilities.
 */
public final class StringUtils {

    public static final String EMPTY_STRING = "";

    private StringUtils() {
    }

    public static boolean isNullOrEmpty(@Nullable String str) {
        if (str == null) {
            return true;
        }

        for (int i = 0; i < str.length(); i++) {
            if (!Character.isWhitespace(str.charAt(i))) {
                return false;
            }
        }

        return true;
    }
}

File: android/app/src/main/java/com/texasgamer/zephyr/util/StringUtils.kt
package com.texasgamer.zephyr.util

/**
 * String utilities.
 */
object StringUtils {
    const val EMPTY_STRING = ""

    @JvmStatic
    fun isNullOrEmpty(str: String?): Boolean {
        if (str == null) {
            return true
        }
        for (i in 0 until str.length) {
            if (!Character.isWhitespace(str[i])) {
                return false
            }
        }
        return true
    }
}